src/desktop-trampoline.c

```diff
@@ -44,6 +45,42 @@ int isValidEnvVar(char *env) {
   return 0;
 }
 
+/**
+ * Reads a string from the socket, reading first 2 bytes to get its length and
+ * then the string itself.
+ */
+ssize_t readDelimitedString(SOCKET socket, char *buffer, size_t bufferSize) {
```

Comment:
This function is basically used to read the `stdout` and `stderr` messages sent from the server.